src/Microsoft.DotNet.Build.Tasks/GenerateDepsJson.cs

```diff
@@ -110,52 +110,56 @@ public override bool Execute()
                 }
             }
 
-            targetsSection.Remove(runtimeTarget);
-            newDepsJson.Remove("libraries");
+            string platformPackageName = "";
+            foreach (var package in targetsSection[runtimeTarget].Children())
+            {
+                JProperty packageProperty = package as JProperty;
+                if (packageProperty != null && packageProperty.Name.Contains($"runtime.{rid}.Microsoft.NETCore.App"))
+                {
+                    platformPackageName = packageProperty.Name;
+                    break;
+                }
+            }
+
+            if (string.IsNullOrEmpty(platformPackageName))
+                Log.LogError($"Unable to find platform package that matches runtime.{rid}.Microsoft.NETCore.App when generating deps.json file.");
+
+            JObject platformPackage = (JObject)targetsSection[runtimeTarget][platformPackageName];
+            platformPackage.Remove("runtime");
+            platformPackage.Remove("native");
+            JObject platformRuntime = new JObject();
+            JObject platformNative = new JObject();
+
             foreach (var assembly in assemblyNames)
             {
-                JObject runtimes = new JObject();
-                JObject runtimeLocation = new JObject();
-                string key = $"{assembly.AssemblyName.Name}/{assembly.AssemblyName.Version.Major}.{assembly.AssemblyName.Version.Minor}.{assembly.AssemblyName.Version.Build}";
-                runtimeLocation.Add(assembly.FileName, new JObject());
-                runtimes.Add("runtime", runtimeLocation);
                 try
                 {
-                    targetValue.Add(key, runtimes);
+                    platformRuntime.Add($"runtimes/{rid}/lib/netcoreapp2.0/{assembly.FileName}", new JObject());
                 }
                 catch (System.ArgumentException)
                 {
 
                 }
+            }
+            // Add mscorlib and System facades to platformRuntime
+            platformRuntime.Add($"runtimes/{rid}/lib/netcoreapp2.0/mscorlib.dll", new JObject());
```

Comment:
Why are these special? They should be in the shared framework.

Reply:
Because these two are the manual shims that get built *after* src.builds is done, which means that they won't be present in the folder by the time that we run this task. That is why manually I added the two shims into the json. You do bring up a good point which is the case of a full rebuild scenario, I believe it will fail given that the platformRuntime will already have a mscorlib and System.dll assemblies. I'll fix this.